### ISSUE TITLE
WMS Option detectRetina

### ISSUE THREAD

**Issue by Poldy1705** [ptv-logistics/Leaflet.NonTiledLayer#26]
Great work!

Is there a possibility to implement the WMS Option for mobile devices : detectRetina: true.

Because the SingleTiles looking very blurry.

**Comment by oliverheilig** [comment]
Yes, it would be possible to query an image twice the size of the actual window with a detectRetina option. However this would also require some kind of dpi parameter for the WMS. Otherwise this would also result in small and unreadable texts or icons. and i don't think there is a standard for this (i.e. WMS always assumes 96 dpi).

It would be no problem for aerial maps or if you're implementing you own rendering service. I will check this out.

**Comment by oliverheilig** [comment]
Found there is a parameter &format_options=dpi:xxx for WMS
https://docs.geoserver.org/latest/en/user/services/wms/vendor.html

But this doesn't seem to be used by any WMS i've tested. But i could add the size-doubling as in
https://github.com/Leaflet/Leaflet/blob/master/src/layer/tile/TileLayer.WMS.js#L83-L86

**Comment by Poldy1705** [comment]
Sorry about the delayed answer. 
The size doubling as in TileLayer WMS is probably the solution for the problem and it would be very nice  if you could add it to your code.
There would be then a parameter like detectretina?
How ist it solved by the core in leaflet or for example in the qwc2 project? https://github.com/qgis/qwc2-demo-app

**Comment by oliverheilig** [comment]
OK, 

i have added a **detectRetina** option to the NonTiledLayer base class, you can set it also for the WMS layer. It just doubles the image size if the option is set and the browser is in retina mode. The issue with this option is that the reference dpi for WMS is 90, so labels and icons will be rendered way too small (see pic 1). It's not an issue for dpi invariant imagery, like satellite images (see pic 2). This is why the option is only set for the 'airmass' layer in the demo.

Afaik there is no standard WMS query parameter for dpi, the option i have mentioned in my post above seems to be specific to GeoServer. So you have to detect the 'retina mode' and set a specific url option by yourself, if you have a service that can deliver specific resolutions.

![image](https://user-images.githubusercontent.com/958768/71257083-102e5e80-2333-11ea-80d6-991e5c932ed1.png)

![image](https://user-images.githubusercontent.com/958768/71257255-8632c580-2333-11ea-96fb-8beb2dbb6e88.png)
